app/routes.py

```diff
@@ -1 +1,254 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint,jsonify, abort,make_response,request
+from app import db
+import requests
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import os
+
+task_list_bp = Blueprint("tasks", __name__, url_prefix ="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix = "/goals")
+
+@task_list_bp.route("", methods = ["POST"])
+def create_tasks():
+    request_body = request.get_json()
+    print(request_body)
+    print("************")
+    
+    if (not "title" in request_body) or (not "description" in request_body):
+        return{
+            "details":"Invalid data"
+        }, 400
+    try:
+        new_task = Task.from_dict(request_body)
+        db.session.add(new_task)
+        db.session.commit()
+
+        #message = f"Task {new_task.title} successfully created"
+        return make_response(jsonify({"task":new_task.task_dict()}), 201)
+    
+    except KeyError as e:
+        abort(make_response("Invalid request. Missing required value: {e}"), 400)
+
+@task_list_bp.route("/<id>", methods = ["GET"])
+def get_one_saved_task(id):
+    task = validate_task(id)
+    return jsonify({"task":task.task_dict()}), 200
+ 
+@task_list_bp.route("", methods = ["GET"])
+def get_all_saved_tasks():
+    sort_query=request.args.get("sort")
+    tasks_query=Task.query
+    
+    if sort_query =="asc":
+        tasks_query = Task.query.order_by(Task.title.asc())
+    elif sort_query =="desc":
+        tasks_query = Task.query.order_by(Task.title.desc())
+    
+    tasks = tasks_query.all()
+
+    tasks_response =[task.task_dict() for task in tasks]
+
+    return (jsonify(tasks_response)),200
+        
+    
+    
+def validate_task(task_id):
+    try:
+        task_id = int(task_id)
+    except:
+        abort(make_response({"message":f"task {task_id} invalid"}, 400))
+
+    
+    task = Task.query.get(task_id)
+  
+    if not task:
+        abort(make_response({"message": f"task {task_id} not found"}, 404))
+
+    return task
+
+
+@task_list_bp.route("/<id>", methods = ["PUT"])
+def update_task(id):
+    task = validate_task(id)
+    request_body = request.get_json()
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    
+    db.session.commit()
+    response_body = {"task":task.task_dict()}
+    
+    
+    return make_response(jsonify(response_body), 200)
```

Comment:
We need to have error handling for a request that uses the request body to create a response (like you have in the POST route). If request_body doesn't have a key "title" then line 75 would throw an unhandled exception.